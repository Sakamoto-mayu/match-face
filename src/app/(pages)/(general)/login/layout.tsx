export const metadata = {
  title: "ログイン",
  description: "一般ユーザーログインページ",
};

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return <>{children}</>;
}
