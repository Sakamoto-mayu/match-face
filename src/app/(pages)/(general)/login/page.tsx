import Input from '@/components/ui/Input';
import OrangeButton from '@/components/ui/button/OrangeButton';
import Link from 'next/link';

const LoginPage = () => {
  return (
    <div className="flex flex-col">
      <h1 className=" text-orange font-normal">Match Face</h1>
      <label htmlFor="userId" className="">
        ユーザーID
      </label>
      <Input id="userId" />
      <label htmlFor="password" className="">
        パスワード
      </label>
      <Input id="password" />
      <OrangeButton label="ログイン" />
      <Link href="/remind" className="text-blue-500">
        パスワードを忘れた
      </Link>
      <Link href="/admin/login" className="text-blue-500">
        管理者ログイン
      </Link>
    </div>
  );
};

export default LoginPage;
