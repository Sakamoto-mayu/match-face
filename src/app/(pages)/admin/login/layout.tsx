export const metadata = {
  title: "ログイン",
  description: "管理者ログインページ",
};

export default function LoginLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return <>{children}</>;
}
