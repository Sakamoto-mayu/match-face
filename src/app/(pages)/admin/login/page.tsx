'use client';
import Input from '@/components/ui/Input';
import SiteTitle from '@/components/ui/SiteTitle';
import OrangeButton from '@/components/ui/button/OrangeButton';
import users from '@/const/login';
import Link from 'next/link';
import { useRouter } from 'next/navigation';
import { FormEvent, useState } from 'react';

const AdminLoginPage = () => {
  const router = useRouter();
  const [userId, setUserId] = useState<string>();
  const [password, setPassword] = useState<string>();
  // const [userIdError, setUserIdError] = useState<boolean>(false);
  const [userIdBlankError, setUserIdBlankError] = useState<boolean>(false);
  // const [passwordError, setPasswordError] = useState<boolean>(false);
  const [passwordBlankError, setPasswordBlankError] = useState<boolean>(false);

  // ダミーデータ
  const adminCorrectUser = users[2];
  // console.log('adminCorrectUser', adminCorrectUser);

  // todo:（仮）ログイン認証チェック(入力欄がブランクの時のみエラー感知)
  const checkLogin = (event: FormEvent) => {
    event.preventDefault();
    if (!userId && !password) {
      setUserIdBlankError(true);
      setPasswordBlankError(true);
    } else if (!userId && password) {
      setUserIdBlankError(true);
      setPasswordBlankError(false);
    } else if (userId && !password) {
      setUserIdBlankError(false);
      setPasswordBlankError(true);
    } else if (
      userId === adminCorrectUser.user_id &&
      password === adminCorrectUser.password
    ) {
      router.push('/admin');
    }
  };

  return (
    <>
      <form onSubmit={checkLogin}>
        <div className="flex flex-col items-center justify-center h-screen">
          <div className="flex flex-col items-center m-3">
            <SiteTitle className="m-3" />
            <h2>管理者用ログイン</h2>
          </div>

          <div className="mt-2 mb-2">
            <label htmlFor="userId" className="">
              ユーザーID
            </label>
            <Input
              id="userId"
              className="w-96 h-8 mt-2"
              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                setUserId(e.target.value)
              }
            />
            {userIdBlankError && (
              <p className="text-red">※ユーザーIDを入力してください</p>
            )}
          </div>
          <div>
            <label htmlFor="password" className="">
              パスワード
            </label>
            <Input
              id="password"
              className="w-96 h-8 mt-2"
              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                setPassword(e.target.value)
              }
            />
            {passwordBlankError && (
              <p className="text-red">※パスワードを入力してください</p>
            )}
          </div>

          <OrangeButton
            label="ログイン"
            className="mt-10 mb-4 w-48 rounded-none"
            type="submit"
          />
          <Link href="/remind" className=" text-blue">
            パスワードを忘れた
          </Link>
          <Link href="/login" className="text-blue pt-16">
            一般ログイン
          </Link>
        </div>
      </form>
    </>
  );
};

export default AdminLoginPage;
