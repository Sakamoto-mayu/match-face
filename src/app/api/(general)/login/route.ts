import { cookies } from 'next/headers';
import { NextResponse } from 'next/server';

export async function POST(request: Request) {
  const res = await request.json();
  const userId = res.userId;
  const password = res.password;
  const response = await fetch(
    `http://localhost:8080/qa_system_api/users/${userId}?password=${password}`
  );
  const data = await response.json();
  cookies().set({
    name: 'userId',
    value: `${userId}`,
    httpOnly: true,
    secure: true,
    path: '/',
  });
  return NextResponse.json(data);
}
