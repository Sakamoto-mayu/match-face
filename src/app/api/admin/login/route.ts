import { cookies } from 'next/headers';
import { NextResponse } from 'next/server';

export async function POST(request: Request) {
  const res = await request.json();
  const administratorId = res.administratorId;
  const password = res.password;
  try {
    const response = await fetch(
      `${process.env.BE_URL}/administrators/${administratorId}?password=${password}`
    );
    const data = await response.json();
    cookies().delete('userId');
    if (data.administrator.id && data.administrator.length !== 0) {
      cookies().set({
        name: 'administratorId',
        value: `${data.administrator.id}`,
        httpOnly: true,
        secure: true,
        path: '/',
        maxAge: 60 * 60 * 24,
      });
    }
    return NextResponse.json(data);
  } catch (error) {
    throw new Error('api error');
  }
}
