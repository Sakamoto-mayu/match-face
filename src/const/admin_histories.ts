export const skills = [
    {id:1,skill:"JavaScript",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,skill:"TypeScript",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,skill:"React",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,skill:"Next.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,skill:"Vue.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:6,skill:"Angular",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:7,skill:"Nuxt.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:8,skill:"jQuery",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:9,skill:"Redux",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:10,skill:"vuex",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:11,skill:"MaterialUI",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:12,skill:"tailwind",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:13,skill:"Bootstrap",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:14,skill:"Vercel",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const departments = [
    {id:1,name:"FR",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:2,name:"Java",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:3,name:"PHP",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:4,name:"CL",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:5,name:"ML",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:6,name:"QA",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
]

export const projects = [
    {id:1,name:"案件A",detail:"案件Aの説明",enterprise_id:1,department_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,name:"案件B",detail:"案件Bの説明",enterprise_id:1,department_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,name:"案件C",detail:"案件Cの説明",enterprise_id:1,department_id:2,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,name:"案件D",detail:"案件Dの説明",enterprise_id:1,department_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,name:"案件E",detail:"案件Eの説明",enterprise_id:1,department_id:2,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
]

export const project_skills = [
    {id:1,project_id:1,skill_id:[1,2,3],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,project_id:2,skill_id:[4,5],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,project_id:3,skill_id:[8,13],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,project_id:4,skill_id:[1,2,5],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,project_id:5,skill_id:[2,3,4,6],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const answer_requests = [
    {id:1,user_id:[1,2],administrator_id:5,project_id:1,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-10T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,user_id:[1,2],administrator_id:5,project_id:2,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-10T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,user_id:[3],administrator_id:5,project_id:3,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-15T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,user_id:[1,2,4],administrator_id:5,project_id:4,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-15T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,user_id:[4],administrator_id:5,project_id:5,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-08-10T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const answers = [
    {id:1,context:'回答が入ります1',question_id:1,answer_request_id:1,user_id:1,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,context:'回答が入ります2',question_id:1,answer_request_id:1,user_id:2,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
    {id:3,context:'回答が入ります3',question_id:2,answer_request_id:2,user_id:1,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-13T16:52:46.053Z'},
    {id:4,context:'回答が入ります4',question_id:2,answer_request_id:2,user_id:2,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,context:'回答が入ります5',question_id:3,answer_request_id:3,user_id:3,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:6,context:'回答が入ります6',question_id:4,answer_request_id:4,user_id:1,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
    {id:7,context:'回答が入ります7',question_id:4,answer_request_id:4,user_id:2,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
    {id:8,context:'回答が入ります8',question_id:4,answer_request_id:4,user_id:4,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:9,context:'回答が入ります9',question_id:5,answer_request_id:5,user_id:4,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
]

export const users = [
    {id:1,name:'田中太郎',password:'tanaka',email:'tanaka@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,name:'山田次郎',password:'yamada',email:'yamada@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,name:'佐藤三郎',password:'sato',email:'sato@example.com',hire_date:'2023-01-01',department_id:2,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,name:'鈴木四郎',password:'suzuki',email:'suzuki@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,name:'高橋五郎',password:'takahashi',email:'takahashi@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]
