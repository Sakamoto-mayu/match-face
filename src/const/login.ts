type User = {
  id: number;
  name: string;
  password: string;
  email: string;
  hire_date: string;
  department_id: number;
  status_id: number;
  created_user: string;
  created_at: string;
  update_user: string;
  update_at: string;
};

type Users = User[];

const users: Users = [
  {
    id: 1111,
    name: 'ラクス太郎',
    password: 'password#1',
    email: 'rakus@example.com',
    hire_date: '2023-01-01',
    department_id: 1,
    status_id: 1,
    created_user: 'ラクス責任者',
    created_at: '2022-12-30',
    update_user: 'ラクス責任者',
    update_at: '2023-01-02',
  },
  {
    id: 2222,
    name: 'ラクス次郎',
    password: 'password#2',
    email: 'rakus2@example.com',
    hire_date: '2023-01-01',
    department_id: 2,
    status_id: 2,
    created_user: 'ラクス責任者',
    created_at: '2022-12-30',
    update_user: 'ラクス責任者',
    update_at: '2023-01-02',
  },
  {
    id: 3333,
    name: 'ラクス三郎',
    password: 'password#3',
    email: 'rakus3@example.com',
    hire_date: '2023-01-01',
    department_id: 3,
    status_id: 3,
    created_user: 'ラクス責任者',
    created_at: '2022-12-30',
    update_user: 'ラクス責任者',
    update_at: '2023-01-02',
  },
];

export default users;
