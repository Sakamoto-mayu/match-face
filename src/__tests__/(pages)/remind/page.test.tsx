import RemindPage from '@/app/(pages)/remind/page';
import { findByText, render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { AppRouterContextProviderMock } from '@/__tests__/test_utils/app-router-context-provider-mock';
import { dummyUser, users } from '@/const/login';

const push = jest.fn();
const user = userEvent.setup();

// alertのモック（修正するかも）

window.alert = jest.fn();

// ダミーデータ
const registeredUser = users[0];
const notRegisteredUser = dummyUser[0];

describe('スナップショット', () => {
  test('スナップショット', () => {
    const { container } = render(
      <AppRouterContextProviderMock router={{ push }}>
        <RemindPage />
      </AppRouterContextProviderMock>
    );
    expect(container).toMatchSnapshot();
  });
});

describe('ページがレンダリングされた時 ', () => {
  beforeEach(() => {
    render(
      <AppRouterContextProviderMock router={{ push }}>
        <RemindPage />
      </AppRouterContextProviderMock>
    );
  });
  test('"一般ログイン"のリンクのurlが"/login"であること', () => {
    const loginLink = screen.getByRole('link', { name: '一般ログイン' });
    expect(loginLink).toHaveAttribute('href', '/login');
  });
});

describe('初めてOrangeButtonがクリックされた時', () => {
  beforeEach(() => {
    render(
      <AppRouterContextProviderMock router={{ push }}>
        <RemindPage />
      </AppRouterContextProviderMock>
    );
  });

  describe('メールアドレスが未入力の時', () => {
    test('"※メールアドレスを入力してください。"が表示されること。', async () => {
      const sendButton = screen.getByRole('button', {
        name: '再設定用メール送信',
      });
      await user.click(sendButton);
      expect(await screen.findByText('※メールアドレスを入力してください。'));
    });
  });

  describe('登録されていないメールアドレスが入力されている時', () => {
    test('"※入力されたメールアドレスが正しくありません。"が表示されること。', async () => {
      const emailInput = screen.getByLabelText('メールアドレス');
      await user.type(emailInput, `${notRegisteredUser.email}`);

      const sendButton = screen.getByRole('button', {
        name: '再設定用メール送信',
      });
      await user.click(sendButton);

      expect(
        await screen.findByText('※入力されたメールアドレスが正しくありません。')
      );
    });
  });
});

describe('エラーメッセージが表示されている時', () => {
  beforeEach(() => {
    render(
      <AppRouterContextProviderMock router={{ push }}>
        <RemindPage />
      </AppRouterContextProviderMock>
    );
  });

  describe('"※メールアドレスを入力してください。"が表示されている時', () => {
    test('登録されているメールアドレスが入力されている場合、"リマインドメールが送信されました。"のアラートが表示、"※メールアドレスを入力してください。"が消えログイン画面に遷移すること', async () => {
      // エラーメッセージの表示
      const sendButton = screen.getByRole('button', {
        name: '再設定用メール送信',
      });
      await user.click(sendButton);
      expect(await screen.findByText('※メールアドレスを入力してください。'));

      // 登録されているメールアドレスを入力
      const emailInput = screen.getByLabelText('メールアドレス');
      await user.type(emailInput, `${registeredUser.email}`);
      // ボタンをクリック（2回目）
      await user.click(sendButton);

      // アラートが表示
      expect(window.alert).toHaveBeenCalledWith(
        'リマインドメールが送信されました。'
      );

      // メッセージが消えている
      expect(
        await waitFor(() =>
          screen.queryByText('※メールアドレスを入力してください。')
        )
      ).toBeNull();

      // 一般ログイン画面へ遷移
      expect(push).toHaveBeenCalledWith('/login');
    });

    test('登録されていないメールアドレスが入力されている場合、"※メールアドレスを入力してください。"が消え、"※入力されたメールアドレスが正しくありません。"が表示されること', async () => {
      // エラーメッセージの表示
      const sendButton = screen.getByRole('button', {
        name: '再設定用メール送信',
      });
      await user.click(sendButton);
      expect(await screen.findByText('※メールアドレスを入力してください。'));

      // 登録されていないメールアドレスを入力
      const emailInput = screen.getByLabelText('メールアドレス');
      await user.type(emailInput, `${notRegisteredUser.email}`);

      await user.click(sendButton);

      // エラーメッセージが切り替わる
      expect(
        await waitFor(() => {
          screen.queryByText('※メールアドレスを入力してください。');
        })
      ).toBeUndefined();
      expect(
        await screen.findByText('※入力されたメールアドレスが正しくありません。')
      );
    });
  });

  describe('"※入力されたメールアドレスが正しくありません。"が表示されている時', () => {
    test('登録されているメールアドレスが入力されている場合、"※入力されたメールアドレスが正しくありません。"が消え、ログイン画面に遷移すること', async () => {
      // エラーメッセージの表示
      const emailInput = screen.getByLabelText('メールアドレス');
      await user.type(emailInput, `${notRegisteredUser.email}`);

      const sendButton = screen.getByRole('button', {
        name: '再設定用メール送信',
      });
      await user.click(sendButton);
      expect(
        await screen.findByText('※入力されたメールアドレスが正しくありません。')
      );

      // 登録されているメールアドレスを入力
      await user.clear(emailInput);
      await user.type(emailInput, `${registeredUser.email}`);

      await user.click(sendButton);

      // アラートが表示
      expect(window.alert).toHaveBeenCalledWith(
        'リマインドメールが送信されました。'
      );

      // メッセージが消えている
      expect(
        await waitFor(() =>
          screen.queryByText('※入力されたメールアドレスが正しくありません。')
        )
      ).toBeNull();

      // 一般ログイン画面へ遷移
      expect(push).toHaveBeenCalledWith('/login');
    });
  });
});
