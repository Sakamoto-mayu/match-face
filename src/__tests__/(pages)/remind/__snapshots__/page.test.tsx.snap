// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スナップショット スナップショット 1`] = `
<div>
  <form>
    <div
      class="flex flex-col items-center justify-center h-screen"
    >
      <h1
        class="text-orange text-4xl font-semibold m-8"
      >
        Match Face
      </h1>
      <span
        class="mb-7"
      >
        パスワード再設定用のURLを登録メールアドレスに送信します。
      </span>
      <div
        class=""
      >
        <label
          for="email"
        >
          メールアドレス
          <div>
            <input
              class=" w-96 h-10 mt-2 border border-black"
              id="email"
              name="email"
              type="text"
            />
          </div>
        </label>
      </div>
      <div>
        <button
          class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white p-2 h-auto text-lg mt-14 mb-4 w-56 rounded-none"
          type="submit"
        >
          再設定用メール送信
        </button>
      </div>
      <a
        class="text-blue"
        href="/login"
      >
        一般ログイン
      </a>
    </div>
  </form>
</div>
`;
